### PR TITLE
Crank Charger

### PR DESCRIPTION
**Pull request by Kurfursten**
- Emergency toolboxes now also contain a screwdriver, wirecutter, and crank charger.
- Cutting the ID wire on an APC will disengage the coverlock (but allow interface access, as pulsing does)
- If the crank charger is loaded with a power cell, clicking it with itself will charge it (10 per click).

Video (one day): http://a.uguu.se/hcclji.webm

> Changes since video: 25 charge per click lowered to 10; spammy message removed, you can now view charge percentage with examine; Fixed a bug where the handle didn't move when clicked to charge

Balance considerations
- Screwdrivers and wirecutters fall in demand
- Power outages can be combated on a local level without breaking into engineering
- Battery changes bring an interesting dynamic to gameplay - they force total darkness in the area while it's being cranked up because the cell must be removed. While cranking, one is potentially vulnerable.
- Making it easier to retrieve power cells may result in an increase in ghetto chemistry (due to the ease of access to cells)
- Crank chargers may be beneficial to engineers repairing an area which has lost power after an explosion, or engineers which are building new rooms in space not linked to the main grid
- Due to the hassle of restoring power in this way and the ease of setting up most engines, likely won't impact engineering's role in the station significantly

### CODE REVIEW

**Comment by Exxion** [comment]
I have two suggestions.
1. Add a delay so someone with a rapid click macro can't charge a cell faster than a powered charger.
2. Change the `if(stored)` in `attack_hand()` to `if(stored && user.get_inactive_hand() == src)` to avoid the shuffling you had to do in the video to get the thing out of the toolbox.

**Comment by Kurfursten** [comment]
Those are good suggestions, though the delay should be very small, something like 1, because while I would want to prevent someone from using a clicking program/macro, I would still want to allow people to charge pretty quickly by hand. The advantage of a stationary charger is that you can leave it and go do other things, this is something you have to devote your whole focus to, and usually the power is out while you're doing it (as you have removed the cell), so I'm okay with it being competitive in speed.

**Comment by Exxion** [comment]
Yeah, 1 would probably be good. What would be bad is the essentially zero delay that is on, for example, rotating chairs. It's fine on rotating chairs because that has no real impact on gameplay, but being able to charge a battery that quickly might be a bit OP.

**Comment by ComicIronic** [comment]
1. How is this superior to ordering new cells via cargo or printing them in science
2. I suggest you make this a chance to spawn in toolboxes, because emergency toolboxes tend to have few supplies in them.

**Comment by Kurfursten** [comment]
@ComicIronic It's not superior to ordering new cells or printing them in science.

I was thinking about moving them out of the emergency toolbox into a new "blackout supplies" box that would be in one of the random O2 locker types.

**Comment by dylanstrategie** [comment]
Only problem, if it wasn't fixed yet, is adding a delay, otherwise you're going to end up with macro binds that will put the Singularity to shame

**Comment by PJB3005** [comment]
Dat sprite though

**Comment by Exxion** [comment]
Oh my god
I just realized how good this would be for exploring in a spacepod

**Comment by PJB3005** [comment]
>exploring in a space pod

What the hell are you exploring anyways, anything you can "eplore" is boring as shit.

**Comment by Exxion** [comment]
Salvage shuttle?
Vox outpost?
Abandoned ship with a laser gun on it?

**Comment by PJB3005** [comment]
It's not really exploring, it's power/metagaming because let's be honest, *everybody* knows where the loot is in space

**Comment by Exxion** [comment]
Yes, but you still have to move around until you find the right Z-level

**Comment by PJB3005** [comment]
GPS, done.

**Comment by Exxion** [comment]
But you still have to move
I was just saying this would be helpful to recharge a spacepod in deep space

**Comment by Kurfursten** [comment]
I wrote this from a hotel room. Moved the crank + supplies to a blackout response kit (20% chance for an O2 locker to be of this type). It also has instructions!

This is for real my last submission for a week, I will be totally cut off from the internet from here on.

File: code/datums/wires/apc.dm [hunk 1]
```diff
@@ -57,6 +57,11 @@ var/const/APC_WIRE_AI_CONTROL = 8
 	var/obj/machinery/power/apc/A = holder
 
 	switch(index)
+		if(APC_WIRE_IDSCAN)
```
Comment (ComicIronic) [inline code/datums/wires/apc.dm]: Unless you check mended, it will do this even if you fix the wire.

File: code/game/machinery/cell_charger.dm [hunk 2]
```diff
@@ -140,3 +140,65 @@
...
+	var/obj/item/weapon/cell/stored = null
+	var/state = 0 //0 if up, 1 if down; only used for icons
+
+/obj/item/device/crank_charger/update_icon()
+	if(stored)
+		icon_state = "crankcharger[state ? "-1" : "-0"]"
+	else
+		icon_state = "crankcharger"
+
+/obj/item/device/crank_charger/examine(mob/user)
```
Comment (ComicIronic) [inline code/game/machinery/cell_charger.dm]: Inherit the parent for this.

**Comment by Kurfursten** [comment]
While I did get a cool icon from anonymous for this one, I just don't see enough real demand for this item after extensive review and talking to players.

**Comment by Exxion** [comment]
That's too bad. I really wanted this, even if only for spacepod ~~exploration~~ powergaming.

**Pull request closed without merging**